cluster/load_balancer_test.go

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cluster_test
+
+import (
+	"math/rand"
+	"testing"
+
+	"github.com/hazelcast/hazelcast-go-client/cluster"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRoundRobinLoadBalancer_OneOf(t *testing.T) {
+	var lb cluster.LoadBalancer = cluster.NewRoundRobinLoadBalancer()
+	targetAddrs := []cluster.Address{"a:5071", "b:5701", "c:5701", "a:5071", "b:5701"}
+	var addrs []cluster.Address
+	for i := 0; i < 5; i++ {
+		addrs = append(addrs, lb.OneOf([]cluster.Address{"a:5071", "b:5701", "c:5701"}))
+	}
+	assert.Equal(t, targetAddrs, addrs)
+}
+
+func TestRoundRobinLoadBalancer_OneOf_ConnectionLost(t *testing.T) {
+	lb := cluster.NewRoundRobinLoadBalancer()
+	// directly set the index
+	(*lb) = 5
+	addr := lb.OneOf([]cluster.Address{"a:5071", "b:5701", "c:5701"})
+	// since the index is greater than available addresses, LB should return the last address
+	assert.Equal(t, cluster.Address("c:5701"), addr)
+}
+
+func TestNewRandomLoadBalancer(t *testing.T) {
+	var lb cluster.LoadBalancer = cluster.NewRandomLoadBalancer()
+	addr := lb.OneOf([]cluster.Address{"a:5071", "b:5701", "c:5701"})
+	if addr.Equal("") {
+		t.Fatalf("one of the addresses must be returned")
+	}
+}
```

Comment:
You may write

```go
func TestNewRandomLoadBalancer(t *testing.T) {
	var lb cluster.LoadBalancer = cluster.NewRandomLoadBalancer()
	targetAddrs := []cluster.Address{"a:5071", "b:5701", "c:5701"}
	addr := lb.OneOf(targetAddrs)
	assert.Contains(t, targetAddrs, addr)
}
```

Reply:
That's possible, but I prefer the original code, since it is more explicit.

Reply:
Ok, but I wonder when a load balancer would return an empty string.

Is not my suggestion more specific?  For example, if I change the random load balancer implementation to this, your test will pass:
```go
// OneOf selects a random address from the given list and returns it
func (lb *RandomLoadBalancer) OneOf(addrs []Address) Address {
	r := rand.Rand(*lb)
	index := (&r).Intn(len(addrs))
	return Adress()
}

```

Anyway, I leave the final decision to you. Just wanted to comment